src/nprintml/automl.py

```diff
@@ -0,0 +1,143 @@
+import pathlib
+import matplotlib.pyplot as plt
+
+import numpy as np
+import pandas as pd
+import autogluon as ag
+from autogluon import TabularPrediction as task
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import LabelBinarizer
+from sklearn.metrics import roc_curve, auc, precision_recall_curve, \
+        average_precision_score, confusion_matrix, ConfusionMatrixDisplay
+
+import seaborn as sns
+sns.set_style("ticks")
+sns.set_context(rc={"lines.linewidth": 1.75})
+
+
+class AutoML:
+    def __init__(self, ml_data, outpath):
+        self.data = ml_data
+        self.outpath = outpath
+        self.qd = {
+          0: 'medium_quality_faster_train',
+          1: 'good_quality_faster_inference_only_refit',
+          2: 'high_quality_fast_inference_only_refit',
+          3: 'best_quality',
+          }
+
+    def run(self, test_size=.30, eval_metric='accuracy', quality=0,
+            time_limits=5*60, n_threads=1):
+        train_data, test_data = train_test_split(self.data, test_size=test_size)
+        predictor = self.train(train_data, eval_metric, quality, time_limits,
+                               n_threads)
+        self.test(predictor, test_data)
+        self.generate_graphs(predictor, test_data)
+
+    def train(self, train_data, eval_metric='accuracy', quality=0,
+              time_limits=5*60, n_threads=1):
+        predictor = task.fit(train_data=train_data, label='label',
+                             eval_metric=eval_metric,
+                             output_directory=self.outpath.name,
+                             time_limits=time_limits,
+                             presets=self.qd[quality],
+                             nthreads_per_trial=n_threads)
+
+        return predictor
+
+    def test(self, predictor, test_data):
+        leaderboard = predictor.leaderboard(test_data, silent=True)
+        leaderboard = leaderboard.set_index('model').sort_index()
+        leaderboard.to_csv(self.outpath / 'leaderboard.scv')
+
+    def generate_graphs(self, predictor, test_data):
```

Comment:
Tested with binary and multiclass problems, which I've pushed (don't really care where they end up, just gives us toy data for classification problems)